qiita_db/base.py

```diff
@@ -256,21 +277,20 @@ def check_status(self, status, exclude=False):
         Table setup:
         foo: foo_status_id  ----> foo_status: foo_status_id, status
         """
-        # Check that self._table is actually defined
-        self._check_subclass()
-
-        # Get all available statuses
-        conn_handler = SQLConnectionHandler()
-
-        statuses = [x[0] for x in conn_handler.execute_fetchall(
-            "SELECT DISTINCT status FROM qiita.{0}_status".format(self._table),
-            (self._id, ))]
+        trans = Transaction("check_status_%s_%s"
+                            % (self.__class__.__name__, self._id))
+        with trans:
+            sql = "SELECT DISTINCT status FROM qiita.{0}_status".format(
+                self._table)
+            trans.add(sql, [self._id])
+            statuses = [x[0] for x in trans.execute()[-1]]
+
+            # Check that all the provided statuses are valid statuses
+            if set(status).difference(statuses):
+                raise ValueError("%s are not valid status values"
```

Comment:
I know you didn't really change this code, but can you add this exception to the raises section and format the set using `', '.join(...)`?